### ISSUE TITLE
Nullable parameters for built-in types

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#2488]
### Description of the issue

When nullable parameters are defined in a module and the consuming module does not pass through a value (resulting in a null) an expansion error is raised.

_The problem is also experienced with custom types, however there is some additional complexity so a separate issue will track the case for custom types #2489._

### Error messages

`The parameter named 'minTLSVersion' was not set or a defaultValue was defined.`

### Reproduction

```bicep
param minTLSVersion string?

resource storage 'Microsoft.Storage/storageAccounts@2023-01-01' = {
  name: 'test'
  #disable-next-line no-loc-expr-outside-params
  location: resourceGroup().location
  sku: {
    name: 'Standard_LRS'
  }
  kind: 'StorageV2'
  properties: {
    minimumTlsVersion: minTLSVersion ?? 'TLS1_2'
  }
}
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.30.2

### Additional context

Raised by @AlexanderSehr in microsoft/PSRule#1640
